docs/faq.rst

```diff
@@ -78,6 +80,54 @@ If the journal you are publishing in requires you to use ``textsc`` for software
 
         \newcommand{\pyhf}{\textsc{pyhf}}
 
+Why use Python?
+~~~~~~~~~~~~~~~
+
+As of the late 2010's Python is widely considered the lingua franca of machine learning
+libraries, and is sufficiently high-level and expressive for physicists of various computational
+skill backgrounds to use.
+Using Python as the language for development allows for the distribution of the software
+--- as both source files and binary distributions --- through the Python Package Index (PyPI)
+and Conda-forge, which significantly lowers the barrier for use as compared to ``C++``.
+Additionally, a 2017 `DIANA/HEP <https://diana-hep.org/>`_ study :cite:`faq-feickert-diana-fellowship-report`
+demonstrated the graph structure and automatic differentiation abilities of machine learning
+frameworks allowed them to be quite effective tools for statistical fits.
+As the frameworks considered in this study (TensorFlow, PyTorch, MXNet) all provided
+low-level Python APIs to the libraries this made Python an obvious choice for a common
+high-level control language.
+Given all these considerations, Python was chosen as the development language.
+
+How did ``pyhf`` get started?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In 2017 Lukas Heinrich was discussing with colleauge Holger Schulz how it would be convienent
+to share and produce statistical results from LHC experiements if they were able to be
+created with tools that didn't require the large ``C++`` dependencies and tooling expertise as
+:math:`\HiFa{}`.
+Around the same time that Lukas began thinking on these ideas, Matthew Feickert was working on
+`a DIANA/HEP fellowship <https://twitter.com/SMUPhysics/status/861584474638766080>`_ with
+Kyle Cranmer (co-author of :math:`\HiFa{}`) to study if the graph structure and automatic
+differentiation abilities of machine learning frameworks would allow them to be effective
+tools for statistical fits.
+Lukas would give helpful friendly advice on Matthew's project and one night [1]_ over dinner
+in CERN's R1 cafeteria the two were discussing the idea of implimenting :math:`\HiFa{}`
+in Python using machine learning libraries to drive the computation.
+Continuing the discussion in Lukas's office, Lukas showed Matthew that the core statistical
+machinery could be implimented rather succinctly, and that night
+`proceeded to do so <https://github.com/scikit-hep/pyhf/commit/fd32503fb760f070a4047cb867757458b1687599>`_
+dubbing the project ``pyhf``.
+
+.. tweet:: https://twitter.com/lukasheinrich_/status/956809112674885632
+
+Matthew joined him on the project to begin development and by April 2018 Giordon Stark had
+learned about the project and began making contributions, quickly becoming
```

Comment:
> I can't remember too much... but I know we ended up in a Table in R1 for a while late at night just working on something related to MBJ + pyhf at the time, and that's probably what was I guess my ramp-up meeting to pyhf then.

Okay. I had this bit in about April as the first PR from you was PR #104 on 2018-04-10, but @kratsg please feel free to edit any and all of this to fill things in more. Additionally, if @lukasheinrich had more memories then those should go in as well. :+1: